From f383834e37ca279c3bd43f7483bc89c86bdaa1ca Mon Sep 17 00:00:00 2001
From: Jan-Michael Brummer <jan-michael.brummer1@volkswagen.de>
Date: Wed, 2 Oct 2024 16:37:27 +0200
Subject: [PATCH] onedrive: Use presentation id as host / user

Currently the drive is shown as unique id and therefore isn't very
userfriendly on cli. Switch to make use of the presentation id (mail)
just like it is done by Google backend to simplify it.
---
 daemon/gvfsbackendonedrive.c | 9 ++++++---
 meson.build                  | 2 +-
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/daemon/gvfsbackendonedrive.c b/daemon/gvfsbackendonedrive.c
index c8317cb7..4f7b6d3a 100644
--- a/daemon/gvfsbackendonedrive.c
+++ b/daemon/gvfsbackendonedrive.c
@@ -1148,6 +1148,7 @@ g_vfs_backend_onedrive_mount (GVfsBackend  *_self,
   g_autolist (MsgDrive) drives = NULL;
   GList *l = NULL;
   const char *host = NULL;
+  const char *user = NULL;
 
   g_debug ("+ mount\n");
 
@@ -1159,7 +1160,8 @@ g_vfs_backend_onedrive_mount (GVfsBackend  *_self,
     }
 
   host = g_mount_spec_get (spec, "host");
-  self->account_identity = g_strdup (host);
+  user = g_mount_spec_get (spec, "user");
+  self->account_identity = g_strconcat (user, "@", host, NULL);
 
   accounts = goa_client_get_accounts (self->client);
   for (l = accounts; l != NULL; l = l->next)
@@ -1170,7 +1172,7 @@ g_vfs_backend_onedrive_mount (GVfsBackend  *_self,
       const char *provider_type = NULL;
 
       account = goa_object_get_account (object);
-      account_identity = goa_account_get_identity (account);
+      account_identity = goa_account_get_presentation_identity (account);
       provider_type = goa_account_get_provider_type (account);
 
       if (g_strcmp0 (provider_type, "ms_graph") == 0 &&
@@ -1254,7 +1256,8 @@ g_vfs_backend_onedrive_mount (GVfsBackend  *_self,
   g_vfs_backend_set_default_location (_self, msg_drive_item_get_name (self->home));
 
   real_mount_spec = g_mount_spec_new ("onedrive");
-  g_mount_spec_set (real_mount_spec, "host", self->account_identity);
+  g_mount_spec_set (real_mount_spec, "host", host);
+  g_mount_spec_set (real_mount_spec, "user", user);
   g_vfs_backend_set_mount_spec (_self, real_mount_spec);
   g_mount_spec_unref (real_mount_spec);
 
diff --git a/meson.build b/meson.build
index 22b58405..395f8cc9 100644
--- a/meson.build
+++ b/meson.build
@@ -368,7 +368,7 @@ endif
 # *** Check if we should build with GOA volume monitor ***
 enable_goa = get_option('goa')
 if enable_goa
-  goa_dep = dependency('goa-1.0', version: '>= 3.17.1')
+  goa_dep = dependency('goa-1.0', version: '>= 3.53.1')
 endif
 
 # *** Check for GNOME Keyring ***
-- 
2.46.2

